npm/rollup-dev-server/src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { EventEmitter } from 'events'
+import { debug as debugFn } from 'debug'
+import { start as createDevServer } from './startServer'
+import { Server } from 'http'
+import { RollupOptions } from 'rollup'
+const debug = debugFn('cypress:rollup-dev-server:rollup')
+
+interface Options {
+  specs: Cypress.Cypress['spec'][] // Why isn't this working? It works for webpack-dev-server
+  config: Record<string, string>
+  devServerEvents: EventEmitter
+  [key: string]: unknown
+}
+
+export interface StartDevServer {
+  /* this is the Cypress options object */
+  options: Options
+  rollupConfig?: RollupOptions // TODO: user's rollup configuration.
+}
+
+export interface ResolvedDevServerConfig {
+  port: number
+  server: Server
+}
+
+export async function startDevServer (startDevServerArgs: StartDevServer): Promise<ResolvedDevServerConfig> {
+  const rollupDevServer = await createDevServer(startDevServerArgs)
+
+  return new Promise(async (resolve) => {
+    const port = 3000
```

Comment:
Can we do a random port...? Should be able to pass in `port: 0` to do this.

Reply:
port: 0 does not work for nollup apparently so I just made a random number instead.